ledger/src/blockstore_db.rs

```diff
@@ -254,6 +254,18 @@ struct Rocks {
     write_batch_perf_status: PerfSamplingStatus,
 }
 
+macro_rules! rocks_try {
```

Comment:
why macro is needed? i think generic wrapper `fn` should suffice at quick glance.

Reply:
`fn` will work as well, but since we are doing this only for rare errors, `macro_rules!` should have less performance impact on the normal case.

Reply:
hmm, i don't think using `macro_rules!` will bring better performance here... could you elaborate a bit? I'm currently thinking the `fn` version would surely be inlined, just like the macro one.